### PR TITLE
add imagecodecs

### PR DESCRIPTION
**Pull request by ocefpaf**
See https://pypi.org/project/imagecodecs/

Should help with: https://github.com/conda-forge/tifffile-feedstock/pull/33

### CODE REVIEW

**Comment by marcelotrevisani** [comment]
I think It needs ``numpy`` on ``host``
https://github.com/scossu/imagecodecs/blob/10a433cf1ae794eb8980850ed601f97810d07ece/setup.py#L173

**Comment by marcelotrevisani** [comment]
Should we try to use ``--universal``?

File: recipes/imagecodecs/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+source:
+  url: https://pypi.io/packages/source/i/imagecodecs/imagecodecs-{{ version }}.tar.gz
+  sha256: 037e3fde4592fffdb4b8235170180d0d0ba01e4640d9660360dcc62f062e2bd1
+
+build:
+  number: 0
+  entry_points:
+    - imagecodecs=imagecodecs.__main__:main
+  script: "{{ PYTHON }} -m pip install . -vv"
```
Comment (marcelotrevisani) [inline recipes/imagecodecs/meta.yaml]: Maybe?
```suggestion
  script: "{{ PYTHON }} -m pip install . -vv --install-option="--universal""
```

File: recipes/imagecodecs/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  run:
+    - python
+    - {{ pin_compatible('numpy') }}
+
+test:
+  imports:
+    - imagecodecs
+  commands:
+    - imagecodecs --help
+  requires:
```
Comment (marcelotrevisani) [inline recipes/imagecodecs/meta.yaml]:
```suggestion
```

**Comment by ocefpaf** [comment]
This would require heavy patching to find the proper c-libs. Giving up on this one for now.

```python
    include_dirs.extend(
        ['/usr/include/jxrlib',
         '/usr/include/openjpeg-2.1',
         '/usr/include/openjpeg-2.2',
         '/usr/include/openjpeg-2.3'])
```

File: recipes/imagecodecs/meta.yaml (hunk 2 above)
Comment (ocefpaf) [reply recipes/imagecodecs/meta.yaml]: This is still a test. I was adding the c-libs one-by-one here.

File: recipes/imagecodecs/meta.yaml (hunk 1 above)
Comment (ocefpaf) [reply recipes/imagecodecs/meta.yaml]: I don't think so. It needed `numpy 1.14.*` and like #10200 it needed the pin somehow.

**Comment by ericpre** [comment]
For info, good progress was made at #6842.

**Pull request closed without merging**